packages/components/src/components/hds/modal/index.hbs

```diff
@@ -3,18 +3,35 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: MPL-2.0
 }}
-<dialog
+<Hds::DialogPrimitive::Wrapper
   class={{this.classNames}}
   ...attributes
   aria-labelledby={{this.id}}
   {{did-insert this.didInsert}}
   {{will-destroy this.willDestroyNode}}
   {{focus-trap isActive=this.isOpen focusTrapOptions=(hash onDeactivate=this.onDismiss clickOutsideDeactivates=true)}}
 >
-  {{yield (hash Header=(component "hds/modal/header" id=this.id onDismiss=this.onDismiss))}}
-  {{yield (hash Body=(component "hds/modal/body"))}}
-  {{yield (hash Footer=(component "hds/modal/footer" onDismiss=this.onDismiss))}}
-</dialog>
+  <:header>
+    {{yield
+      (hash
+        Header=(component
+          "hds/dialog-primitive/header" id=this.id onDismiss=this.onDismiss contextualClassPrefix="hds-modal"
```

Comment:
Should this be: `contextualClassPrefix="hds-modal__header"`?

Reply:
no, it's different: it's not the usual contextualClass, it's just the prefix (see the code in the Dropdown::Header)